boxoffice/models/line_item.py

```diff
@@ -255,10 +256,30 @@ def sales_by_date(dates, user_tz, item_ids):
     return date_sales
 
 
-def sales_delta(user_tz, item_ids):
+def calculate_weekly_sales(item_collection_ids, user_tz, year):
     """
-    Calculates the percentage difference in sales between today and yesterday
+    Calculates sales per week for items in the given set of item_collection_ids in a given year,
+    in the user's timezone.
     """
+    week_sales = db.session.query('sales_week', 'sum').from_statement('''SELECT DATE_TRUNC('week', ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)
+        AS sales_week, SUM(final_amount)
+        FROM line_item INNER JOIN item on line_item.item_id = item.id
+        WHERE status=:status AND item_collection_id IN :item_collection_ids AND
+        DATE_TRUNC('year', ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone) = :year GROUP BY sales_week ORDER BY sales_week;
```

Comment:
This comparision for `year` forces it to scan every single row in the db. You need to be checking `ordered_at` >= earliest point in the year, in UTC, and < same for next year.